### PR TITLE
Enhancement for logs

### PR DESCRIPTION
**Pull request by AmirDoreh**
This pr offers:

- some comments enhancements
- refactoring codes
- initializing siteCache to prevent nil map access.

### CODE REVIEW

File: log/logger.go [hunk 1]
```diff
@@ -173,8 +173,8 @@ func (l *logger) Write(level slog.Level, msg string, attrs ...any) {
 	l.inner.Handler().Handle(context.Background(), r)
 }
 
-func (l *logger) Log(level slog.Level, msg string, attrs ...any) {
-	l.Write(level, msg, attrs...)
+func (l *logger) Log(level slog.Level, msg string, ctx ...interface{}) {
```
Comment (MariusVanDerWijden) [inline log/logger.go:176]: Why do you prefer interface to any here?

File: log/handler_glog.go [hunk 2]
```diff
@@ -184,20 +186,21 @@ func (h *GlogHandler) Handle(_ context.Context, r slog.Record) error {
...
 		h.lock.Lock()
+		defer h.lock.Unlock()
 
 		fs := runtime.CallersFrames([]uintptr{r.PC})
 		frame, _ := fs.Next()
 
 		for _, rule := range h.patterns {
 			if rule.pattern.MatchString(fmt.Sprintf("+%s", frame.File)) {
 				h.siteCache[r.PC], lvl, ok = rule.level, rule.level, true
+				break
```
Comment (MariusVanDerWijden) [inline log/handler_glog.go:197]: Doesn't this change the logic by only matching on the first occurrence instead of every occurrence?

File: log/handler_glog.go [hunk 3]
```diff
@@ -184,20 +186,21 @@ func (h *GlogHandler) Handle(_ context.Context, r slog.Record) error {
 	// If we didn't cache the callsite yet, calculate it
 	if !ok {
 		h.lock.Lock()
+		defer h.lock.Unlock()
```
Comment (MariusVanDerWijden) [inline log/handler_glog.go:189]: Afaict this will lock the handler even for the h.origin.Handle call in line 206 (203 in original) which means we hold the lock longer than we need to

File: log/format.go [hunk 4]
```diff
@@ -30,70 +30,42 @@ type TerminalStringer interface {
...
-			buf.WriteString(color)
-			buf.Write(appendEscapeString(buf.AvailableBuffer(), attr.Key))
-			buf.WriteString("\x1b[0m=")
-		} else {
-			buf.Write(appendEscapeString(buf.AvailableBuffer(), attr.Key))
-			buf.WriteByte('=')
-		}
+		buf.WriteString(color)
+		buf.Write(appendEscapeString(buf.AvailableBuffer(), attr.Key))
+		buf.WriteString("\x1b[0m=")
```
Comment (MariusVanDerWijden) [inline log/format.go:66]: Pretty sure that this is wrong, previously we wrote `=` when color was "", now its "\x1b[0m="

File: log/format.go [hunk 5]
```diff
@@ -30,70 +30,42 @@ type TerminalStringer interface {
...
-	if color != "" { // Start color
-		b.WriteString(color)
-		b.WriteString(LevelAlignedString(r.Level))
-		b.WriteString("\x1b[0m")
-	} else {
-		b.WriteString(LevelAlignedString(r.Level))
-	}
+	b.WriteString(color)
+	b.WriteString(LevelAlignedString(r.Level))
+	b.WriteString("\x1b[0m")
```
Comment (holiman) [inline log/format.go:45]: This should only be written if we're using terminal colors

File: log/handler.go [hunk 6]
```diff
@@ -37,16 +37,13 @@ func (h *discardHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
...
-	wr       io.Writer
-	lvl      slog.Level
-	useColor bool
-	attrs    []slog.Attr
-	// fieldPadding is a map with maximum field value lengths seen until now
-	// to allow padding log contexts in a bit smarter way.
-	fieldPadding map[string]int
-
-	buf []byte
+	mu       sync.Mutex   // Mutex for thread safety
```
Comment (holiman) [inline log/handler.go:40]: "Mutex for thread safety" is a moot description.
The fact that it's a mutex is already given, and mutexes are used for "thread safety", or rather, to handle mutually exclusive accesses in a multi-threaded context. 

If you want to put a meaningful description, you should instead say something like "mu is used to prevent multu-threaded access on the buf field" or something like that, which describes specifically what fields the `mu` guards.

File: log/handler_glog.go [hunk 7]
```diff
@@ -139,8 +126,24 @@ func (h *GlogHandler) Vmodule(ruleset string) error {
 	return nil
 }
 
+// compilePattern converts a vmodule pattern to a regular expression string.
+func compilePattern(pattern string) string {
```
Comment (holiman) [inline log/handler_glog.go:130]: what's the benefit of moving this out?

**Review by holiman — COMMENTED** [review]
I don't really see a theme here, looks a bit like a few random changes here and there in the log package. But none of it really meaningful. Am I missing something?

File: log/handler_glog.go [hunk 8]
```diff
@@ -168,7 +170,7 @@ func (h *GlogHandler) WithGroup(name string) slog.Handler {
 	panic("not implemented")
 }
 
-// Log implements Handler.Log, filtering a log record through the global, local
```
Comment (BabyHalimao) [inline log/handler_glog.go:171]: This has been fixed in #29907 please revert this changes

**Review by BabyHalimao — CHANGES_REQUESTED** [review]

**Comment by holiman** [comment]
In the past many months we have had a lot of issues with people opening PRs that are - for the most part - not useful. These include typo fixes, performance optimisations in non-hot code paths, variable renames, error message tweaks, etc. Whilst these PRs aren't unilaterally bad or worthless, most people are not opening them to make our codebase better, rather they are opening to farm commits for airdrops.

From our perspective, these commits often take a non-negligible effort to review and merge, because they keep touching sensitive code, that we need to be extra careful to understand. Even trivial fixes require attention and keep distracting us from focusing on developing and maintaining Geth.

We've decided to be more agressive in rejecting outside contributions that do not make a meaningful improvement:

- Issues flagged by linters or third party tools should be discussed on a GitHub issue first to see if it makes sense to fix them, and if so, the correct way is to add a new linter step to our CIs so that issues are caught when being added, not kept being fixed in perpetuity.
- Optimisations (especially when they lower code readability) should be done only in hot paths where performance actually matters, and even in those cases, there should be some benchmarks (ideally from the existing ones) highlighting their benefit.
- As for typos, variable renames, error rewordings and such, should be opened as part of larger meaningful changes, not as tiny standalone PRs.

We appreciate the enthusiasm and effort of opening a PR against Geth, but due to the significant abuse against our time, we'll be closing PRs of the above type for the foreseeable future. We apologise if someone's genuine effort gets caught up in this agressive stance.
